scripts/superdesk-authoring/authoring.js

```diff
@@ -236,13 +235,23 @@ import 'angular-history/history.js';
             } else {
                 return api.find('archive', _id, {embedded: {lock_user: 1}})
                 .then(function _lock(item) {
-                    if (!read_only && lock.isLocked(item)) {  // Check if item is still editable
-                        item._locked = true;
+                    if (read_only) {
+                        item._locked = lock.isLockedInCurrentSession(item);
                         item._editable = false;
                         return $q.when(item);
-                    } else {
-                        item._editable = !read_only;
-                        return lock.lock(item);
+                    } else { // we want to lock
+                        if (lock.isLocked(item)) { // is locked by someone else
```

Comment:
yep, this needs more work, but don't want to change it now just before next release